LEAF_Request_Portal/auth_cookie/index.php

```diff
@@ -16,106 +17,98 @@
 //$protocol = isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on' ? 'https://' : 'http://';
 $protocol = 'https://';
 
-if (isset($_COOKIE['REMOTE_USER']))
-{
-    $redirect = '';
-    if (isset($_GET['r']))
-    {
-        $redirect = $protocol . HTTP_HOST . base64_decode($_GET['r']);
-    }
-    else
-    {
-        $redirect = $protocol . HTTP_HOST . dirname($_SERVER['PHP_SELF']) . '/../';
-    }
+if (isset($_COOKIE['REMOTE_USER'])) {
+    $defaultRedirect = $protocol . HTTP_HOST . dirname($_SERVER['PHP_SELF']) . '/.../';
+    $redirect = Security::getSafeRedirectFromRequest(HTTP_HOST, $defaultRedirect, $protocol);
 
-    $user = decryptUser($_COOKIE['REMOTE_USER']);
+    $user = Security::decryptUser($_COOKIE['REMOTE_USER'], CIPHER_KEY);
 
     // see if user is valid
     $vars = array(':userName' => $user);
-    $res = $oc_db->prepared_query('SELECT * FROM employee
-                                            WHERE userName=:userName
-                                            AND deleted=0', $vars);
+    $sql = 'SELECT *
```

Comment:
The select * caught my attention.  It looks like all that's being used from the return is a true/false if the person exists in the database.